### PR TITLE
59 add localization comments to the dashboardview

### LINKED ISSUE #59: FEATURE - Add Localization Comments to the `DashboardView`
## Describe the feature you to improve
* Add localization comments to `DashboardView`

## Describe alternatives you've considered
* N/A

## Additional context
* Describe what the string is about, a button to do XYZ, a tab item label showing ABC info. The localization comment is a short string to help people localize the strings in the app

### PR DESCRIPTION
**Pull request by hartti**
# What it Does
* Closes #59
* Add localization comments for all UI strings
* The Localizable file now shows all the new comments to help the people working localizations to understand what each string means

# How I Tested
* Build the application
* Localizable file should show all the new string comments

# Notes
* There are still strings without comments from other Views

# Screenshot
Screenshot 2023-10-29 at 23 23 23

### CODE REVIEW

File: Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift [hunk 1]
```diff
@@ -25,15 +25,17 @@ struct DashboardView: View {
             List {
                 ForEach(viewModel.searchedEvents) { event in
                     VStack(alignment: .leading, spacing: 8) {
-                        Text(event.title)
+                      Text(event.title)
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift]: This isn't commented

File: Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift [hunk 2]
```diff
@@ -25,15 +25,17 @@ struct DashboardView: View {
...
+                        Text("For \(event.vehicle.name)",
+                             comment: "Maintenance list item description 'For which vehicle'")
                         
-                        Text("\(event.date.formatted(date: .abbreviated, time: .omitted))")
+                        Text("\(event.date.formatted(date: .abbreviated, time: .omitted))",
+                             comment: "Maintenance list item description 'Date' formatted")
                         
                         if !event.notes.isEmpty {
-                            Text(event.notes)
+                          Text(event.notes)
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift]: This one isn't commented

File: Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift [hunk 3]
```diff
@@ -150,15 +167,20 @@ struct DashboardView: View {
...
+            }
         } message: {
             Text(viewModel.errorMessage)
         }
     }
     
     private var noSearchResultsDescription: Text {
-        Text("There were no maintenance events for '\(viewModel.searchText)'. Try a new search.")
+        Text("There were no maintenance events for '\(viewModel.searchText)'. Try a new search.",
+             comment: "Text shwon when there are no results for maintenance search")
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift:190]: typo here

File: Basic-Car-Maintenance/Shared/Localizable.xcstrings [hunk 4]
```diff
@@ -135,7 +136,7 @@
       }
     },
     "Add" : {
-      "comment" : "Label for submit button on form to add an entry",
+      "comment" : "Label for submit button on form to add an entry\nLabel for adding maintenance event on Dashboard view",
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Localizable.xcstrings]: Note that this one has a `\n` on the comment, which means that this key `Add` has two different string comments, which at the moment causes a conflict.

To fix this you need to either use the same comment for both, or make a new localized string key for `Add` and use your separate comment.

I think making a new one is the best solution

**Review by mikaelacaron — CHANGES_REQUESTED** [review]
Great work!! I have several comments that need resolved please. If you have any questions, you can comment directly on that thread, otherwise click the resolve button once you've addressed my comment. Once you fix everything, please click the re-review button

Also please solve the merge conflict, and fix the indentation for a few spots

Re-review button

File: Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift (hunk 2 above)
Comment (hartti) [reply Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift]: I have to doublecheck this, when I am on my laptop, but I might need to change the line to be Text("\(event.notes)", comment: ...) to be able to add the comment. That is ok, if you want to have the comment. However, there isn't anything to translate, so I am wondering if adding a comment makes any sense?

File: Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift (hunk 3 above)
Comment (hartti) [reply Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift:190]: Dang. Will fix this

File: Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift (hunk 2 above)
Comment (mikaelacaron) [reply Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift]: That's a good point, probably don't need to translate it, no worries, I'm removing this

File: Basic-Car-Maintenance/Shared/Localizable.xcstrings (hunk 4 above)
Comment (hartti) [reply Basic-Car-Maintenance/Shared/Localizable.xcstrings]: Done

**Comment by hartti** [comment]
Uh... where is the re-review button? Newbie is getting a little lost here.. Sorry.

**Comment by mikaelacaron** [comment]
> Uh... where is the re-review button? Newbie is getting a little lost here.. Sorry.

It's at the top right of the page when you're on the conversation tab

File: Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift [hunk 5]
```diff
@@ -55,7 +56,8 @@ struct DashboardView: View {
                             isShowingEditView = true
                         } label: {
                             VStack {
-                                Text("Edit")
+                                Text("Edit",
+                                     comment: "Button label to start editing maintenance")
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift]:
```suggestion
                                Text("Edit",
                                     comment: "Button label to edit this maintenance")
```

File: Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift [hunk 6]
```diff
@@ -30,13 +30,14 @@ struct DashboardView: View {
             List {
                 ForEach(viewModel.searchedEvents) { event in
                     VStack(alignment: .leading, spacing: 8) {
-                        Text(event.title)
+                      Text(event.title)
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift]:
```suggestion
                        Text(event.title)
```

File: Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift [hunk 7]
```diff
@@ -30,13 +30,14 @@ struct DashboardView: View {
...
                             .font(.title3)
                         
-                        Text("\(event.vehicle.name) on \(event.date, formatter: self.eventDateFormat)")
+                        Text("\(event.vehicle.name) on \(event.date, formatter: self.eventDateFormat)",
+                            comment: "Maintenance list item for a vehicle on a date")
                         
                         if !event.notes.isEmpty {
-                            Text(event.notes)
+                          Text(event.notes)
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift]:
```suggestion
                            Text(event.notes)
```

File: Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift [hunk 8]
```diff
@@ -74,7 +76,8 @@ struct DashboardView: View {
                     ProgressView("Loading...")
                 } else {
                     if viewModel.events.isEmpty {
-                        Text("Add your first maintenance")
+                        Text("Add your first maintenance",
+                         comment: "Placeholder text for empty maintenance list")
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift]:
```suggestion
                        Text("Add your first maintenance",
                         comment: "Placeholder text for empty maintenance list prompting the user to add a maintenance event")
```

File: Basic-Car-Maintenance/Shared/Localizable.xcstrings (hunk 4 above)
Comment (mikaelacaron) [reply Basic-Car-Maintenance/Shared/Localizable.xcstrings]: This wasn't resolved, I fixed it

**Review by mikaelacaron — APPROVED** [review]
Great work!! Just a few formatting comments that I'll fix

**Pull request merged**